src/protvista-uniprot.ts

```diff
@@ -209,68 +217,102 @@ class ProtvistaUniprot extends LitElement {
     loadComponent('protvista-uniprot-structure', ProtvistaUniprotStructure);
   }
 
-  _loadData() {
+  async _loadData() {
     const accession = this.accession;
     if (accession && this.config) {
-      const tasks = this.config.categories.map(
-        ({ name, url, adapter, tracks }) => {
-          const urlWithProtein =
-            url.indexOf('{}') >= 0
-              ? url.replace('{}', accession)
-              : `${url}${accession}`;
-          // TODO: remove this conditional setTimeout when InterPro API more stable
-          // NOTE: this is just to ensure the InterPro fetches are enqueued last
-          return sleep(/interpro/.test(urlWithProtein) ? 100 : 0)
-            .then(() => load(urlWithProtein))
-            .then(({ payload }) => {
-              if (!payload) return;
-              const data = adapter ? adapters[adapter](payload) : payload;
-              this.data[name] =
-                adapter === 'protvista-feature-adapter'
-                  ? data.filter(
-                      ({ category }: { category?: string }) =>
-                        !category || category === name
+      // Get the list of unique urls
+      const urls = this.config.categories
+        .map(({ tracks }) => tracks.map(({ data }) => data[0].url))
+        .flat();
+      const uniqueUrls = [...new Set(urls)];
+      // Get the data for all urls and store it
+      for (const url of uniqueUrls) {
+        try {
+          const data = await load(url.replace('{accession}', accession));
+          this.rawData[url] = data.payload;
+        } catch (e) {
+          // TODO handle this better based on error code
+          // Fail silently for now
+        }
+      }
+
+      // Now iterate over tracks and categories, transforming the data
+      // and assigning it as adequate
+      this.config.categories.map(
+        ({ name: categoryName, tracks, trackType }) => {
+          // const categoryData: any = [];
+          const categoryData = tracks.map(
+            ({ data: dataConfig, name: trackName, filter }) => {
+              const { url, adapter } = dataConfig[0]; // TODO handle array
+              const trackData = this.rawData[url] || [];
```

Comment:
Added `[]` as otherwise it wasn't working in when there was no antigen data (see the second example in index.html).

Reply:
I guess then that now you don't need the extra check I've added in 71c6be2 then

Reply:
I guess not but maybe safer to leave it in, I guess eventually we can have all the data types defined properly by the different adapters and simplify a lot of checks.